knowledge-base/editor-add-icons.md

```diff
@@ -0,0 +1,206 @@
+---
+title: Add Icons in Editor
+description: How to allow inserting <i> tags in the Editor for Blazor to render custom icons in the Editor content?
+type: how-to
+page_title: Add Icons in Editor
+slug: editor-kb-add-icons
+position: 
+tags: editor, blazor, icon, schema, prosemirror
+ticketid: 
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Editor for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+This KB article answers the following questions:
+
+* How to add custom icons in the Editor content? When adding an icon such as `<i class="fa-light fa-envelope">`, the icon's HTML disappears upon saving.
+* How can I add an `<i>` element inside the Editor?
+
+## Solution
+
+The built-in ProseMirror schema that the Editor uses does not support an `<i>` tag. To allow adding icons in the Editor content:
```

Comment:
This statement implies that the Editor schema depends on ProseMirror and is outside our control, which is not the case. Consider something like:

```suggestion
The default schema of the Telerik Editor does not include an `<i>` tag, so the Editor strips such elements automatically. To allow adding font icons through `<i>` tags in the Editor content:
```